web-client/src/shared/components/drivers-registration/PersonalInfo/index.test.tsx

```
@@ -0,0 +1,192 @@
+}
+
+
+describe('PersonalInfo', ()=>{
+    const mockOnSubmit = jest.fn()
+    
+    const createTestElements = ()=>{
+        return {
+            firstNameField: screen.getByLabelText('First Name'),
+            lastNameField: screen.getByLabelText('Last Name'),
+            emailField: screen.getByLabelText('Email Address'),
+            phoneField: screen.getByLabelText('Phone Number'),
+            language: screen.getByLabelText('English'),
+            languageOther: screen.getByPlaceholderText('Other'),
+            otherLanguageCheckBox: screen.getByTestId('other-language'),
+            nextBtn: screen.getByText('Next')
+        }
+    }
+
+    describe('when provided information does not meet the criteria', ()=>{
+        render(
+            <Provider store={store}>
+                <PersonalInfoComponent onSubmit={mockOnSubmit}/>
+            </Provider>
+        )
+
+        const {
+            firstNameField,
+            lastNameField,
+            emailField,
+            phoneField,
+            language,
+            otherLanguageCheckBox,
+            languageOther,
+            nextBtn
+        } = createTestElements()
+
+        beforeEach(()=>{
+            // Filling in valid data so each field can be 
+            // tested individually for error and so every 
+            // field doesn't show error on submit
+            UserEvent.type(firstNameField, mockDriverInputData.firstName)
+            UserEvent.type(lastNameField, mockDriverInputData.lastName)
+            UserEvent.type(emailField, mockDriverInputData.email)
+            UserEvent.type(phoneField, mockDriverInputData.phone)
+            UserEvent.click(language)
+        })
+
+        const assertEmptyFieldErrorShouldBeDefined = (field:HTMLElement)=>{
+            fireEvent.change(field, {target:{value: ''}})
+            UserEvent.click(nextBtn)
+
+            const errorMsgElem = screen.getByText('Please fill the field')
+            expect(errorMsgElem).toBeInTheDocument()
```

Perhaps a more concise way of doing this is using the findByText method, which should throw an error if the elem does not exist.